starlite/plugins/base.py

```diff
@@ -91,32 +78,78 @@ class SerializationPluginProtocol(Protocol[ModelT, DataContainerT]):
     __slots__ = ()
 
     @staticmethod
-    def is_plugin_supported_type(value: Any) -> TypeGuard[ModelT]:
+    def container_from_bytes(
+        container_type: type[DataContainerT],
+        data: bytes,
+        media_type: RequestEncodingType | str,
+    ) -> DataContainerT:
+        """Parse an instance of ``container_type`` from raw bytes.
+
+        Args
+            container_type: a container model type
+            data: bytes to be parsed into instance
+            media_type: content encoding of the raw data
+
+        Returns
+            Instance of ``container_type``.
+        """
+        raise NotImplementedError
+
+    @staticmethod
+    def container_array_from_bytes(
+        container_type: type[DataContainerT],
+        data: bytes,
+        media_type: RequestEncodingType | str,
+    ) -> list[DataContainerT]:
+        """Parse an encoded array of objects into a list of ``container_type`` instances.
+
+        Args
+            container_type: a container model type
+            data: bytes to be parsed into instance list
+            media_type: content encoding of the raw data
+
+        Returns
+            List of ``container_type`` instances.
+        """
+        raise NotImplementedError
+
+    @staticmethod
+    def container_to_dict(container_instance: DataContainerT) -> dict[str, Any]:
+        """Convert ``container_instance`` to dict.
+
+        Args
+            container_instance: the container instance
+
+        Returns
+            dict representation of container instance
+        """
+        raise NotImplementedError
+
+    @staticmethod
+    def is_plugin_supported_type(value: Any) -> bool:
```

Comment:
I think this would better be called `supports_type`, as it makes more sense semantically.

`plugin.supports_type(foo)` vs `plugin.is_plugin_supported_type(foo)`. The latter feels a bit redundant.